app/controllers/users/backup_code_setup_controller.rb

```diff
@@ -3,13 +3,17 @@ class BackupCodeSetupController < ApplicationController
     include MfaSetupConcern
     include RememberDeviceConcern
     include SecureHeadersConcern
+    include ReauthenticationRequiredConcern
 
     before_action :authenticate_user!
     before_action :confirm_user_authenticated_for_2fa_setup
     before_action :ensure_backup_codes_in_session, only: %i[continue refreshed]
     before_action :set_backup_code_setup_presenter
     before_action :apply_secure_headers_override
     before_action :authorize_backup_code_disable, only: [:delete]
+    before_action :confirm_recently_authenticated_2fa, if: -> do
+      IdentityConfig.store.reauthentication_for_second_factor_management_enabled
+    end
```

Comment:
Do we / should we assert anything in the controller specs around this expected behavior?

Reply:
The controller currently doesn't have any re-authentication expectations, and the feature flag is intended to preserve current behavior, so in this and the other controllers, the (current) options are disabled or 2FA re-authentication.

We could add tests, but I've erred on keeping the testing of the ReauthenticationRequiredConcern to its own spec to avoid some duplication. There are feature tests around the enabled behavior though.

Reply:
Yeah I guess what I had in mind with controller specs was something like what we do in others with `have_actions` to ensure that the `before_action`s are actually called, but I think feature specs work just as well to provide coverage.

Reply:
Added in d0a6b7251